source/retryable-reads/retryable-reads.rst

```diff
@@ -356,26 +360,43 @@ and reflects the flow described above.
     /* Allow ServerSelectionException to propagate to our caller, which
      * can then assume that no attempts were made. */
     server = selectServer();
-    connection = server.getConnection()   
+
+    /* PoolClearedException indicates the operation did not even attempt to
+     * create a connection, let alone execute the operation. This means we
+     * are always safe to attempt a retry. We do not need to update SDAM,
+     * since whatever error caused the pool to be cleared will do so itself. */
+    try {
+      connection = server.getConnection()
```

Comment:
The psuedocode between retryable-reads and retryable-writes is a bit divergent. Here, the connection check out step is shown explicitly and happens before retryablity support is determined, but in retryable-writes checkout is done as part of execution. I didn't bother trying to unify them in this PR and instead just updated each as necessary to fit the behavioral changes specified in the prose. In DRIVERS-746, we'll likely have to do the full reunification.

In a related note: I think we should consider prioritizing DRIVERS-1401. It's a bit strange that we have error labels for determining retryable writes but don't have the same machinery for reads.